src/mpi4py/MPI.src/asbuffer.pxi

```diff
@@ -75,6 +76,10 @@ cdef int PyMPI_GetBuffer(object obj, Py_buffer *view, int flags) except -1:
             raise
         raise
 
+cdef void PyMPI_ReleaseBuffer(int kind, Py_buffer *view) noexcept:
+    if kind != 0: Py_CLEAR(view.obj)
+    PyBuffer_Release(view)
```

Comment:
I might suggest to make this an if/else clause.  `PyBuffer_Release` shouldn't do anything additional here and that way it is more obvious that this is custom use of `Py_buffer` that comes with custom cleanup.

But the change looks good to me to fix the cupy issue.